sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -108,6 +108,9 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
   }
 
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+    case PhysicalOperation(_, _, DataSourceV2ScanRelation(
```

Comment:
This problem would be solveable with plan costs, but AFAIK, first physical plan is always used for execution.

Reply:
Lets protect code with SQLConf